### PR TITLE
Introduce a workaround for Geckodriver/653

### PR DESCRIPTION
**Pull request by andrewnicols**
This is a workaround for Geckodriver/#653 which is yet to be solved and has been stale for the past year.

When an anchor contains a block element, Firefox is unable to click it.

The suggested workaround is to click on a descendant node instead.

### CODE REVIEW

**Comment by oleg-andreyev** [comment]
@andrewnicols I've rebased your branch on master.

File: src/WebDriver.php [hunk 1]
```diff
@@ -781,7 +782,29 @@ private function clickOnElement(WebDriverElement $element)
             $this->webDriver->action()->moveToElement($element)->perform();
         }
 
-        $element->click();
+        if ($this->browserName === 'firefox') {
```
Comment (oleg-andreyev) [inline src/WebDriver.php:785]: could you please add a testcase?

**Comment by andrewnicols** [comment]
I've created https://github.com/oleg-andreyev/driver-testsuite/pull/3

**Comment by andrewnicols** [comment]
Pinging @oleg-andreyev,

I believe all blockers here should now be lifted once the new test lands.

**Comment by oleg-andreyev** [comment]
Merged testsuite, re-running tests.

**Comment by oleg-andreyev** [comment]
Thanks @andrewnicols 🙏

**Comment by andrewnicols** [comment]
Thanks :)

**Pull request merged**